airflow/security/kerberos.py

```diff
@@ -47,6 +47,10 @@
 log = logging.getLogger(__name__)
 
 
+def get_kerberos_principle(principal: str | None) -> str:
+    return principal or conf.get_mandatory_value("kerberos", "principal").replace("_HOST", get_hostname())
```

Comment:
If this function is to be used elsewhere, it should have a docstring.

Reply:
Yes it will be used from providers.
This PR extracted the core part from https://github.com/apache/airflow/pull/34386

Reply:
@zeotuan can you add it?

Reply:
Sure l'll add it